test/ModifiedFiles.test.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+/* globals describe it */
+const path = require("path");
+const MemoryFs = require("memory-fs");
+const webpack = require("../");
+const fs = require("fs");
+const rimraf = require("rimraf");
+
+const createCompiler = config => {
+	const compiler = webpack(config);
+	compiler.outputFileSystem = new MemoryFs();
+	return compiler;
+};
+
+const tempFolderPath = path.join(__dirname, "temp");
+const tempFilePath = path.join(tempFolderPath, "temp-file.js");
+const tempFile2Path = path.join(tempFolderPath, "temp-file2.js");
+
+const createSingleCompiler = () => {
+	return createCompiler({
+		entry: tempFilePath,
+		watch: true,
+		output: {
+			path: tempFolderPath,
+			filename: "bundle.js"
+		}
+	});
+};
+
+describe("ModifiedFiles", () => {
+	if (process.env.NO_WATCH_TESTS) {
+		it.skip("watch tests excluded", () => {});
+		return;
+	}
+
+	jest.setTimeout(10000);
+
+	function cleanup() {
+		rimraf.sync(tempFolderPath);
+	}
+
+	beforeEach(() => {
```

Comment:
This seem to duplicate a lot of code from `RemovedFiles.test.js`. Maybe it makes sense to merge both files into one.

Reply:
this is a good point, I will merge them into a `ChangesAndRemovals.test.js` file